modules/glib/2.82.2.bcr.2/overlay/glib/BUILD.bazel

```diff
@@ -0,0 +1,314 @@
+    args = [
+        UPSTREAM_VERSION,
+        "visibility-macros",
+        "GLIB",
+        "$(location include_glib-2.0/glib/glib-visibility.h)",
+    ],
+    tool = "//tools:gen-visibility-macros",
+)
+
+cc_library(
+    name = "stub_intl",
+    hdrs = ["stub_libintl/libintl.h"],
+    includes = ["stub_libintl"],
+    visibility = [":__subpackages__"],
+)
+
+GLIB_POSIX_SRCS = [
+    "gspawn-posix.c",
+    "glib-unix.c",
+    "giounix.c",
+]
+
+GLIB_WIN32_SRCS = [
+    "giowin32.c",
+    "gwin32.c",
+    "gspawn-win32-helper.c",
+]
+
+GLIB_INTERNAL_TEXTUAL_HDRS = [
+    "gspawn-win32.c",
+    "gstdio-private.c",
+    "gthread-posix.c",
+    "gthread-win32.c",
+    "gwin32-private.c",
+    "win_iconv.c",
+]
+
+GLIB_TOOL_SRCS = [
+    "gtester.c",
+]
+
+GLIB_SRCS = glob(
+    [
+        "*.c",
+        "deprecated/*.c",
+        "libcharset/*.c",
+        "libcharset/*.h",
+    ],
+    exclude = (
+        GLIB_POSIX_SRCS +
+        GLIB_WIN32_SRCS +
+        GLIB_INTERNAL_TEXTUAL_HDRS +
+        GLIB_TOOL_SRCS
+    ),
+)
+
+cc_library(
+    name = "glib_internal_textual_hdrs",
+    textual_hdrs = GLIB_INTERNAL_TEXTUAL_HDRS,
+)
+
+REPO_ROOT = package_relative_label(":BUILD.bazel").workspace_root
+
+INCLUDE_PREFIX = REPO_ROOT if REPO_ROOT else "."
+
+cc_library(
+    name = "glib",
+    srcs = GLIB_SRCS + select({
+        "@platforms//os:windows": GLIB_WIN32_SRCS,
+        "//conditions:default": GLIB_POSIX_SRCS,
+    }) + PUBLIC_HDRS + INTERNAL_HDRS + select({
+        "@platforms//os:windows": PUBLIC_HDRS_WIN32,
+        "//conditions:default": [],
+    }),
+    hdrs = [
+        ":public_hdr_" + hdr.replace("/", "_")
+        for hdr in PUBLIC_HDRS
+    ] + [
+        ":glib-visibility_h",
+        ":glibconfig_h",
+        ":gversionmacros_h",
+    ] + select({
+        "@platforms//os:windows": [
+            ":public_hdr_" + hdr.replace("/", "_")
+            for hdr in PUBLIC_HDRS_WIN32
+        ],
+        "//conditions:default": [],
+    }),
+    copts = [
+        "-I" + INCLUDE_PREFIX,
+        "-I%s/glib" % (INCLUDE_PREFIX,),
+        "-I%s/glib/deprecated" % (INCLUDE_PREFIX,),
+    ],
+    implementation_deps = [
+        ":glib_internal_textual_hdrs",
+        "//:config_h",
+    ] + select({
+        "@platforms//os:linux": [],
+        "//conditions:default": [":stub_intl"],
+    }),
+    includes = ["include_glib-2.0"],
+    linkopts = select({
+        "@platforms//os:linux": [
+            "-lpthread",
+        ],
+        "@platforms//os:macos": [
+            "-liconv",
+            "-framework",
+            "Carbon",
+        ],
+        "//conditions:default": [],
+    }),
+    linkstatic = True,
+    local_defines = [
+        "_FILE_OFFSET_BITS=64",
+        "G_ENABLE_DEBUG",
+        "_GNU_SOURCE",
+        "G_LOG_DOMAIN='\"GLib\"'",
+        "GLIB_COMPILATION",
+        "GLIB_CHARSETALIAS_DIR='\"/usr/lib\"'",
+    ],
+    visibility = ["//visibility:public"],
+    deps = ["@pcre2"],
```

Comment:
Why was `@pcre2` moved from `implementation_deps=` to `deps=` here? Are there downstream users of glib that expect `#include <glib.h>` to also provide PCRE2 symbols?